File: papers/landscape/verify_from_raw.jl
# Verify paper claims directly from raw JSONL. Julia standard library only.
# Usage: julia verify_from_raw.jl

using JSON, Statistics

claims = JSON.parsefile("claims.json")

readjsonl(f) = [JSON.parse(line) for line in eachline(f)]

v1 = readjsonl("data/leakage_landscape_v1_final.jsonl")
v1_ok = filter(r -> get(r, "status", nothing) == "ok", v1)

v2 = readjsonl("data/leakage_landscape_v2.jsonl")

v3_an = readjsonl("data/v3/v3_an.jsonl")
v3_an_ok = filter(r -> get(r, "v3_status", nothing) == "ok", v3_an)

function dz(vals)
    x = filter(!isnan, vals)
    mean(x) / std(x)
end

function check(name, expected, got; tol=0.002)
    ok = abs(expected - got) <= tol
    println("  $name: expected=$expected, got=$(round(got, digits=4)), $(ok ? "PASS" : "FAIL")")
    ok
end

passed = 0; total = 0

println("=== Dataset counts ===")
global total += 1; global passed += check("n_datasets", claims["n_datasets"], length(v1_ok), tol=0)
global total += 1; global passed += check("corpus.median_n", claims["corpus"]["median_n"],
    median([r["n_rows"] for r in v1_ok]), tol=0)

println("\n=== Class I: Estimation ===")
d = [r["a_lr_gap_diff"] for r in v1_ok if haskey(r, "a_lr_gap_diff") && r["a_lr_gap_diff"] !== nothing]
global total += 1; global passed += check("norm_lr.dz", claims["norm_lr"]["dz"], dz(d))

println("\n=== Class II: Peeking ===")
d = [r["b_infl_k10"] for r in v1_ok if haskey(r, "b_infl_k10") && r["b_infl_k10"] !== nothing]
global total += 1; global passed += check("peek.dz", claims["peek"]["dz"], dz(d))
global total += 1; global passed += check("peek.auc", claims["peek"]["auc"], mean(d), tol=0.001)

println("\n=== Class II: Seed ===")
d = [r["ai_inflation"] for r in v2 if haskey(r, "ai_inflation") && r["ai_inflation"] !== nothing && !isnan(r["ai_inflation"])]
global total += 1; global passed += check("seed.dz", claims["seed"]["dz"], dz(d))

println("\n=== Class II: Screen ===")
d = [r["aq_k1_optimism"] for r in v2 if haskey(r, "aq_k1_optimism") && r["aq_k1_optimism"] !== nothing && !isnan(r["aq_k1_optimism"])]
global total += 1; global passed += check("screen.dz", claims["screen"]["dz"], dz(d))

println("\n=== N-scaling ===")
global total += 1; global passed += check("nscale.n_main", claims["nscale"]["n_main"],
    count(r -> get(r, "an_n_full", 0) == 2000, v3_an_ok), tol=0)
global total += 1; global passed += check("nscale.ext.n_datasets", claims["nscale"]["ext"]["n_datasets"],
    count(r -> get(r, "an_n_full", 0) == 10000, v3_an_ok), tol=0)

println("\n" * "="^40)
println("RESULT: $passed/$total checks passed")
passed == total && println("ALL CLAIMS VERIFIED FROM RAW DATA.")

File: papers/landscape/verify_from_raw.R
# Verify paper claims directly from raw JSONL. Base R only.
# Usage: Rscript verify_from_raw.R

library(jsonlite)

claims <- fromJSON("claims.json")

v1 <- stream_in(file("data/leakage_landscape_v1_final.jsonl"), verbose = FALSE)
v1_ok <- v1[v1$status == "ok", ]

v2 <- stream_in(file("data/leakage_landscape_v2.jsonl"), verbose = FALSE)

v3_an <- stream_in(file("data/v3_an.jsonl"), verbose = FALSE)
v3_an_ok <- v3_an[v3_an$v3_status == "ok", ]

dz <- function(x) {
  x <- x[!is.na(x)]
  mean(x) / sd(x)
}

check <- function(name, expected, got, tol = 0.002) {
  ok <- abs(expected - got) <= tol
  cat(sprintf("  %s: expected=%s, got=%s, %s\n", name, expected, round(got, 4),
              ifelse(ok, "PASS", "FAIL")))
  ok
}

passed <- 0L; total <- 0L

cat("=== Dataset counts ===\n")
total <- total + 1L; passed <- passed + check("n_datasets", claims$n_datasets, nrow(v1_ok), tol = 0)
total <- total + 1L; passed <- passed + check("corpus.median_n", claims$corpus$median_n,
                                                median(v1_ok$n_rows), tol = 0)

cat("\n=== Class I: Estimation ===\n")
d <- v1_ok$a_lr_gap_diff[!is.na(v1_ok$a_lr_gap_diff)]
total <- total + 1L; passed <- passed + check("norm_lr.dz", claims$norm_lr$dz, dz(d))

cat("\n=== Class II: Peeking ===\n")
d <- v1_ok$b_infl_k10[!is.na(v1_ok$b_infl_k10)]
total <- total + 1L; passed <- passed + check("peek.dz", claims$peek$dz, dz(d))
total <- total + 1L; passed <- passed + check("peek.auc", claims$peek$auc, mean(d), tol = 0.001)

cat("\n=== Class II: Seed ===\n")
d <- v2$ai_inflation[!is.na(v2$ai_inflation)]
total <- total + 1L; passed <- passed + check("seed.dz", claims$seed$dz, dz(d))

cat("\n=== Class II: Screen ===\n")
d <- v2$aq_k1_optimism[!is.na(v2$aq_k1_optimism)]
total <- total + 1L; passed <- passed + check("screen.dz", claims$screen$dz, dz(d))

cat("\n=== N-scaling ===\n")
total <- total + 1L; passed <- passed + check("nscale.n_main", claims$nscale$n_main,
                                                sum(v3_an_ok$an_n_full == 2000), tol = 0)
total <- total + 1L; passed <- passed + check("nscale.ext.n_datasets", claims$nscale$ext$n_datasets,
                                                sum(v3_an_ok$an_n_full == 10000), tol = 0)

cat(sprintf("\n%s\nRESULT: %d/%d checks passed\n", strrep("=", 40), passed, total))
if (passed == total) cat("ALL CLAIMS VERIFIED FROM RAW DATA.\n")
